Libraries/LibJS/Interpreter.cpp

```diff
@@ -47,8 +47,47 @@ Interpreter::~Interpreter()
 {
 }
 
+Value Interpreter::destructive_run(Statement& statement, ArgumentVector arguments, ScopeType scope_type)
```

Comment:
Too much copy-pasting here! Can we find a way to share this whole function with `run()` instead of making a copy?

Reply:
Not..really...short of making it a template anyway
The loop's range is different, so we can't just `?:` that

splitting it into parts and only keeping the loop is also rather silly.